### PR TITLE
Issues with Javamail test failures

### PR DESCRIPTION
**Pull request by uberskigeek**

### CODE REVIEW

**Comment by uberskigeek** [comment]
fixes #9300

**Comment by uberskigeek** [comment]
#build

File: dev/com.ibm.ws.javamail.1.6/src/com/ibm/ws/javamail/internal/injection/MailSessionResourceFactoryBuilder.java [hunk 1]
```diff
@@ -72,56 +69,10 @@ public ResourceFactory createResourceFactory(Map<String, Object> props) throws E
...
-        String storeProtocolClass = (String) annotationProps.remove(MailSessionService.STOREPROTOCOLCLASSNAME);
-        String transportProtocolClass = (String) annotationProps.remove(MailSessionService.TRANSPORTPROTOCOLCLASSNAME);
-        String property = (String) annotationProps.remove(MailSessionService.PROPERTY);
-        String mailSessionID = getMailSessionID(application, module, component, jndiName);
-
-        StringBuilder filter = new StringBuilder(FilterUtils.createPropertyFilter(ID, mailSessionID));
-        filter.insert(filter.length() - 1, '*');
-        // Fail if server.xml is already using the id
-        if (!removeExistingConfigurations(filter.toString()))
-            throw new IllegalArgumentException(mailSessionID); // internal error, shouldn't ever have been permitted in server.xml
```
Comment (brideck) [inline dev/com.ibm.ws.javamail.1.6/src/com/ibm/ws/javamail/internal/injection/MailSessionResourceFactoryBuilder.java]: I suspect that the IllegalArgumentException check around the mailSessionID needs to stay in, so at a minimum the old way of handling application, module, component, and jndiName would need to remain.  The new loop should be able to handle all of the rest, though.

Comment (uberskigeek) [reply dev/com.ibm.ws.javamail.1.6/src/com/ibm/ws/javamail/internal/injection/MailSessionResourceFactoryBuilder.java]: think this should fix that.

File: dev/com.ibm.ws.javamail.1.6/src/com/ibm/ws/javamail/internal/injection/MailSessionResourceFactoryBuilder.java [hunk 2]
```diff
@@ -94,34 +81,10 @@ public ResourceFactory createResourceFactory(Map<String, Object> props) throws E
         if (!removeExistingConfigurations(filter.toString()))
             throw new IllegalArgumentException(mailSessionID); // internal error, shouldn't ever have been permitted in server.xml
 
-        mailSessionSvcProps.put(MailSessionService.JNDI_NAME, jndiName);
```
Comment (brideck) [inline dev/com.ibm.ws.javamail.1.6/src/com/ibm/ws/javamail/internal/injection/MailSessionResourceFactoryBuilder.java]: Need to also leave in the puts for the four properties you removed above.  jndiName here...

File: dev/com.ibm.ws.javamail.1.6/src/com/ibm/ws/javamail/internal/injection/MailSessionResourceFactoryBuilder.java [hunk 3]
```diff
@@ -94,34 +81,10 @@ public ResourceFactory createResourceFactory(Map<String, Object> props) throws E
...
-            mailSessionSvcProps.put(MailSessionService.TRANSPORTPROTOCOLCLASSNAME, transportProtocolClass);
-        if (property != null)
-            mailSessionSvcProps.put(MailSessionService.PROPERTY, property);
-
-        if (application != null) {
-            mailSessionSvcProps.put("application", application);
-            if (module != null) {
-                mailSessionSvcProps.put("module", module);
-                if (component != null)
-                    mailSessionSvcProps.put("component", component);
```
Comment (brideck) [inline dev/com.ibm.ws.javamail.1.6/src/com/ibm/ws/javamail/internal/injection/MailSessionResourceFactoryBuilder.java]: ... and application, module, and component here.

File: dev/com.ibm.ws.javamail.1.6/src/com/ibm/ws/javamail/internal/injection/MailSessionResourceFactoryBuilder.java (hunk 2 above)
Comment (uberskigeek) [reply dev/com.ibm.ws.javamail.1.6/src/com/ibm/ws/javamail/internal/injection/MailSessionResourceFactoryBuilder.java]: updated

File: dev/com.ibm.ws.javamail.1.6/src/com/ibm/ws/javamail/internal/injection/MailSessionResourceFactoryBuilder.java (hunk 3 above)
Comment (uberskigeek) [reply dev/com.ibm.ws.javamail.1.6/src/com/ibm/ws/javamail/internal/injection/MailSessionResourceFactoryBuilder.java]: updated

**Pull request merged**